gnd/src/main/java/com/google/android/gnd/ui/home/HomeScreenViewModel.java

```diff
@@ -164,11 +190,29 @@ private boolean shouldShowAddFeatureButton(Loadable<Project> project) {
     return errors;
   }
 
-  public void addFeature(Layer layer, Point point) {
-    getActiveProject()
-        .ifPresent(
-            project ->
-                addFeatureClicks.onNext(featureRepository.newFeature(project, layer, point)));
+  public void addFeature(Project project, Layer layer, Point point) {
+    addFeatureClicks.onNext(featureRepository.newFeature(project, layer, point));
+  }
+
+  public void addPolygonFeature(Project project, Layer layer, List<Point> points) {
+    addPolygonFeatureClicks.onNext(featureRepository
+        .newPolygonFeature(project, layer, ImmutableList.copyOf(points)));
+  }
+
+  public void onAddPolygonBtnClick(Point location) {
+    addPolygonDialogRequests.setValue(Event.create(location));
+  }
+
+  public LiveData<Event<Point>> getShowAddPolyDialogRequests() {
```

Comment:
Poly->Polygon.

What does "poly dialog" refer to?

Also, please use an Observable backed by BehaviorSubject for events instead of Livedata<Event>